### PR TITLE
Add sorting to the columns

### PR DESCRIPTION
**Pull request by buda1994**

### CODE REVIEW

File: src/Grid/DialogDropdown.js [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+  { Value: 'None', Title: 'None' },
+  { Value: 'Equals', Title: 'Equals' },
+  { Value: 'NotEquals', Title: 'Not Equals' }
+];
+
+class DialogDropdown extends React.Component {
+  render(){
+    const value = this.props.value === undefined ? 'None' : this.props.value;
+    
+    if (this.props.columnType === 'string') {
```
Comment (kadosh) [inline src/Grid/DialogDropdown.js]: What about a switch for this case?

File: src/Grid/DialogContent.js [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+          label={'Value'} 
+          classes={this.props.classes} 
+          columnType={this.props.columnType}
+          activeFilter={this.props.activeFilter} 
+          handleDatePicker={this.props.handleDatePicker}
+          handleBooleanDropDown={this.props.handleBooleanDropDown}
+          handleTextFieldChange={this.props.handleTextFieldChange} 
+        />
+        
+        {this.props.operator === 'Between' ?
```
Comment (geoperez) [inline src/Grid/DialogContent.js]: Use short-circuit when the second path is null... For example:

`condition ? <Component /> : null;`

can be coded like:

`condition && <Component />;`

File: src/Grid/DialogDropdown.js (hunk 1 above)
Comment (geoperez) [reply src/Grid/DialogDropdown.js]: Come on, just one return and set a variable for Operators.

File: src/Grid/GridHeader.js [hunk 3]
```diff
@@ -110,144 +93,111 @@ class GridHeader extends React.Component {
...
 
     this.state.dataSource.filter(this.state.rowsPerPage, this.state.page);
   }
 
+  sortHandler = property => {
+    const dataSource = Object.assign([{}], this.state.dataSource);
+
+    dataSource.columns.forEach( (row, i) => {
+      if(row.SortOrder === 1){
+        dataSource.columns[i].SortOrder = -1;
```
Comment (geoperez) [inline src/Grid/GridHeader.js]: `dataSource.columns[i].SortOrder = -1;` is not like `row.SortOrder = -1;`

File: src/Grid/GridHeader.js [hunk 4]
```diff
@@ -110,144 +93,111 @@ class GridHeader extends React.Component {
...
         <TableRow>
           {dataSource.columns.map(column => {
             const render = column.Sortable ?
               (<Tooltip title='Sort' placement='bottom-start' enterDelay={300}>
-                <TableSortLabel
-                  onClick={() => this.sortHandler(column.Label, column.DataType)}
-                >
+                <TableSortLabel onClick={() => this.sortHandler(column.Name)} >
                   {column.Label}
+                  {column.SortOrder === 1 && column.SortDirection === 'Ascending' ?
```
Comment (geoperez) [inline src/Grid/GridHeader.js]: The SortOrder should not be evaluated here... Please check the logic at tubular.

File: src/Grid/GridHeader.js [hunk 5]
```diff
@@ -110,144 +93,111 @@ class GridHeader extends React.Component {
       value2 = 1;
     }
     
-    this.filterHandler(value1, value2);
+    this.filterHandler(value1, value2, true);
   }
 
-  filterHandler = (value1, value2) => {
+  filterHandler = (value1, value2, hasFilter) => {
```
Comment (kadosh) [inline src/Grid/GridHeader.js]: Can we provide a more meaningful name for value1, value2? It is not clear what those are intended for

File: src/Grid/GridHeader.js [hunk 6]
```diff
@@ -63,71 +62,87 @@ class GridHeader extends React.Component {
...
-    const dataSource = Object.assign([{}], this.state.dataSource);
-
-    dataSource.columns.forEach( (row, i) => {
-      if(row.SortOrder === 1){
-        dataSource.columns[i].SortOrder = -1;
-      }
+  sortHandler = (event, property) => {
+    let sortOrder = 1;
 
+    this.state.dataSource.columns.forEach( row => {
```
Comment (kadosh) [inline src/Grid/GridHeader.js]: Please rename that "row" variable, you're doing forEach over columns array, not rows.

File: src/Grid/DialogDropdown.js [hunk 7]
```diff
@@ -32,19 +32,27 @@ const BooleanOperators = [
 class DialogDropdown extends React.Component {
```
Comment (jpalcala) [inline src/Grid/DialogDropdown.js]: Use stateless component

File: src/Grid/GridHeader.js [hunk 8]
```diff
@@ -104,44 +106,110 @@ class GridHeader extends React.Component {
...
+        if(column.SortOrder === -1) {
+          column.SortOrder = sortOrder;
+        }
+        else if(column.SortOrder !== -1 && column.SortDirection === 'Descending'){
+          column.SortOrder = -1;
+        }
+
+        const currentlySortedColumns = array.columns.filter(col => col.SortOrder > 0).sort((a, b) => a.SortOrder === b.SortOrder ? 0 : a.SortOrder > b.SortOrder );
+
+        for (let i = 0; i < currentlySortedColumns.length; i ++) {
```
Comment (kadosh) [inline src/Grid/GridHeader.js]: forEach please

File: src/Grid/GridHeader.js [hunk 9]
```diff
@@ -104,44 +106,110 @@ class GridHeader extends React.Component {
...
+      if(column.Name === property){
+        if(column.SortDirection === 'Descending') {
           sortOrder = -1;
         }
 
-        row.SortOrder = sortOrder;
+        column.SortOrder = sortOrder;
         this.state.dataSource.sort(this.state.rowsPerPage, this.state.page);
-        row.SortDirection = row.SortDirection === 'Ascending' ? 
+        column.SortDirection = column.SortDirection === 'Ascending' ?
```
Comment (kadosh) [inline src/Grid/GridHeader.js]: Please take a look at https://github.com/unosquare/tubular/blob/74d0eee2735dc0b966a4dca893974fa895414a75/src/js/tubular/directives/grid/tbGridCtrl.js#L434

Same validation is there and it's more readable there

File: src/Grid/GridHeader.js [hunk 10]
```diff
@@ -182,6 +182,41 @@ class GridHeader extends React.Component {
...
 
     this.setState({ sortOrder }, this.state.dataSource.sort(this.state.rowsPerPage, this.state.page) );
+  } */
+
+  multiSort = (event, property) => {
+    const array = Object.assign({}, this.state.dataSource);
+    
+    array.columns.forEach( (column, i) => {
+      if(column.Name === property){
+        column.SortDirection = column.SortDirection === 'Ascending' ?
```
Comment (kadosh) [inline src/Grid/GridHeader.js]: Again, this logic can be more readable, check https://github.com/unosquare/tubular/blob/74d0eee2735dc0b966a4dca893974fa895414a75/src/js/tubular/directives/grid/tbGridCtrl.js#L434

**Pull request merged**